core/src/main/resources/db/changelog/changesets/2024_09_30_referenced_questions.yaml

```diff
@@ -0,0 +1,20 @@
+databaseChangeLog:
+  - changeSet:
+      id: "referenced_questions"
```

Comment:
since this is a computed cache-y value that's only used for internal purposes, I'd lean toward having it be a column on Survey directly.  I'd recommend using a `text[]` column (Jdbi will handle autoconverting it into a `List<String>`, the only difficulty is that you have to add the column with sql since `text[]` isn't recognized by liquibase `addColumn` yaml) and just storing a list of `surveyStableId.questionStableId` strings.  That avoids the race conditions in populate and makes the fetching less complicated

Reply:
I think the race condition still exists, as long as we verify on survey creation that the other questions exist. But I didn't know about `text[]`! Great idea.

Reply:
I would be ok skipping the "other question exists" validation.  It might be nice to let survey creators create surveys in either order

Reply:
Fair enough - we'll have to keep an eye out if this parsing starts to surface garbage data, it could just mean we need to more intelligently filter out potential question references.